### PR TITLE
Make kernel connection function synchronous

### PR DESCRIPTION
**Pull request by jasongrout**
Related to work in and built on top of #4697.

### CODE REVIEW

**Comment by jasongrout** [comment]
Rebased on master...

**Comment by jasongrout** [comment]
@blink1073 - do you remember why the kernel connectTo function returns a promise, i.e., is asynchronous? https://github.com/jupyterlab/jupyterlab/blob/98d0671538907952c94c258129da93e1fc6c1a1c/packages/services/src/kernel/default.ts#L1312-L1324

Just checking to make sure I'm not messing things up by not remembering why it is the way it is.

**Comment by jasongrout** [comment]
Tracing it back further, https://github.com/jupyterlab/jupyterlab/blame/09884af82886c630d0f2fa662d6f8a1525613100/packages/services/src/kernel/default.ts#L1343 indicates that the we used to get the kernel model from the server if we didn't have it available locally.

It seems that now the arguments include the model, and that is all we need to create a DefaultKernel. So it does seem that the finalization of the services library was where we basically decided that connectTo did not need to be asynchronous (i.e., didn't need to query the server).

**Comment by blink1073** [comment]
That sounds right to me. I'd say leave it, not worth the major version bump.

**Comment by jasongrout** [comment]
Thanks. We're already doing a major version bump (to make the kernel message processing async and in order), so I was seeing if it was possible to fix this while we're at it.

@ian-r-rose - I think we may be able to get this into beta 3....I'll spend just a little bit more time chasing down the loose ends.

**Comment by jasongrout** [comment]
I think this may actually be good to go for beta 3.

**Comment by jasongrout** [comment]
@ian-r-rose - could you review this?

**Comment by jasongrout** [comment]
Thanks Steve!

File: packages/help-extension/src/index.tsx [hunk 1]
```diff
@@ -172,96 +172,95 @@ function activate(app: JupyterLab, mainMenu: IMainMenu, palette: ICommandPalette
...
+      }
+      commands.addCommand(bannerCommand, {
+        label: `About the ${kernelName} Kernel`,
+        isVisible: usesKernel,
+        isEnabled: usesKernel,
+        execute: () => {
+          // Create the header of the about dialog
+          let headerLogo = (<img src={kernelIconUrl} />);
+          let title = (
+            <span className='jp-About-header'>,
```
Comment (ian-r-rose) [inline packages/help-extension/src/index.tsx]: I think this happened when converting the dialogs to JSX, but I saw it when testing: this JSX has commas in it that it should not have.

Comment (ian-r-rose) [reply packages/help-extension/src/index.tsx]: I went ahead and fixed the issue, hope you don't mind @jasongrout

**Comment by ian-r-rose** [comment]
Looks good, thanks @jasongrout!

**Pull request merged**